### PR TITLE
[WC] Fix ratio_defining_params

### PR DESCRIPTION
**Pull request by l-bat**
### Reason for changes

Fix bug in `_get_ratio_defining_params` method

### Tests

test_shared_gather_all_layers

### CODE REVIEW

File: tests/openvino/native/quantization/test_weights_compression.py [hunk 1]
```diff
@@ -455,6 +455,21 @@ def test_shared_gather(mode):
             assert op.get_element_type() == weight_name_vs_type[op_name]
 
 
+@pytest.mark.parametrize("all_layers", (True, False))
+def test_shared_gather_all_layers(all_layers):
```
Comment (ljaljushkin) [inline tests/openvino/native/quantization/test_weights_compression.py:459]: Did you manage to catch the incorrect behavior with the old code? Seems like this test passed before

**Review by ljaljushkin — CHANGES_REQUESTED** [review]

File: tests/openvino/native/quantization/test_weights_compression.py (hunk 1 above)
Comment (l-bat) [reply tests/openvino/native/quantization/test_weights_compression.py:459]: Looks like the old behavior was also correct. I couldn't find incorrect configuration.

File: nncf/quantization/algorithms/weight_compression/algorithm.py [hunk 2]
```diff
@@ -165,28 +165,28 @@ def _get_ratio_defining_params(
...
-            return list(filter(lambda wp: len(wp.reduction_axes) == 1, all_weight_params))
-
         ratio_defining_params = list(
             filter(
                 lambda wp: wp.node_with_weight.metatype in self._backend_entity.matmul_metatypes,
                 all_weight_params,
             )
         )
 
+        # The last MatMul layer is quantized to 4-bits if all_layers=True or if the layer is shared
```
Comment (ljaljushkin) [inline nncf/quantization/algorithms/weight_compression/algorithm.py:175]: This comment looks wrong.
`if the layer is shared` means that it has the same precision as embedding, and it's 4bit only with `all_layers`

**Review by ljaljushkin — APPROVED** [review]
LGTM, would be nice to correct comment, but it's minor

**Review by ljaljushkin — APPROVED** [review]

**Pull request merged**